etna/transforms/decomposition/change_points_trend.py

```diff
@@ -112,15 +96,10 @@ def fit(self, df: pd.DataFrame) -> "_OneSegmentChangePointsTrendTransform":
         -------
         :
         """
-        series = df.loc[df[self.in_column].first_valid_index() : df[self.in_column].last_valid_index(), self.in_column]
-        if series.isnull().values.any():
-            raise ValueError("The input column contains NaNs in the middle of the series! Try to use the imputer.")
-        change_points = _find_change_points_segment(
-            series=series, change_point_model=self.change_point_model, **self.change_point_model_predict_params
-        )
-        self.intervals = self._build_trend_intervals(change_points=change_points)
-        self.per_interval_models = self._init_detrend_models(intervals=self.intervals)
-        self._fit_per_interval_model(series=series)
+        super(_OneSegmentChangePointsTrendTransform, self).fit(df=df)
+
+        self.per_interval_models = self._init_detrend_models(intervals=self.intervals)  # type: ignore
+        self._fit_per_interval_model(series=self.series)
```

Comment:
how about to make this method common for all change-point transforms?

Reply:
It not used in segmentation. Also self.per_interval_models will have to be declared in base class, although such an attribute is not needed in segmentation.